### PR TITLE
ci: update to latest version of c8 for realtime versioned test coverage

### PR DESCRIPTION
**Pull request by jmartin4563**
## Proposed Release Notes
* Update to latest version of c8 for realtime versioned test coverage

## Details
Noticed that this repo does its versioned tests different than the other external repos while working a different ticket: now that a new version of c8 has been published, I took the liberty of updating this repo to be more in line with the other externals and take advantage of the new c8 `--merge-async` functionality.

* Removed nightly run as it's no longer necessary
* Removed scheduled run of CI, we don't do that for other external repos and we already have the agent nightly that will run all the externals
* Removed logic switch for changing sampling rate depending on branch
* Bumped to latest version of c8 with @bizob2828 's changes for `--merge-async`
* Update versioned test runner to specify `--merge-async` when invoking c8

### CODE REVIEW

File: bin/run-versioned-tests.sh [hunk 1]
```diff
@@ -3,13 +3,9 @@
 set -x
 VERSIONED_MODE="${VERSIONED_MODE:---minor}"
 SAMPLES="${SAMPLES:-15}"
+C8_REPORTER="${C8_REPORTER:-lcov}"
```
Comment (bizob2828) [inline bin/run-versioned-tests.sh:6]: I don't see you overriding this in CI with `lcovonly`. I assume that was your intention correct?

Comment (jmartin4563) [reply bin/run-versioned-tests.sh:6]: Ah, yes, I need to update the workflow to include the env var. I also noticed that we're not currently providing the minor/major env var, so we're always running with the `--minor` flag even on PRs, do you think I should change it so we run on PRs with `--major`?

Comment (bizob2828) [reply bin/run-versioned-tests.sh:6]: That was done purely to cut down on time in the main agent.  I'll leave that up to you to decide

Comment (jmartin4563) [reply bin/run-versioned-tests.sh:6]: I ended up adding the VERSIONED_MODE flag in after all, without it the PR build was [20 minutes](https://github.com/newrelic/newrelic-node-apollo-server-plugin/actions/runs/5137181985/jobs/9244937127), which seemed a bit long imo.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**